MotorI2C/motorI2C.cpp

```diff
@@ -0,0 +1,61 @@
+#include "MotorI2C.h"
+#include "SmartMotor.h"
+
+MotorI2C::MotorI2C(int slaveAddress) {
+  _slaveAddress = slaveAddress;
+  _leftMotor = nullptr;  // Pointer to SmartMotor instance for left motor
+  _rightMotor = nullptr; // Pointer to SmartMotor instance for right motor
+  _newDataAvailable = false;
+  _lastCommandTime = 0;
+}
+
+// Setup I2C communication and initialize motors
+void MotorI2C::setupI2C(SmartMotor* leftMotor, SmartMotor* rightMotor) {
+  _leftMotor = leftMotor;
+  _rightMotor = rightMotor;
+  Wire.begin(_slaveAddress);  // Initialize the I2C as a slave
+  Wire.onReceive(receiveData);  // Set the receive function to handle incoming data
+  Wire.onRequest(sendTelemetry); // Set the request function to handle telemetry data
```

Comment:
Be very careful with this! Initialising the `Wire` interface inside an "hidden" function risks breakage that's very difficult to debug.

It's always best to initialise every hardware interface in the Arduino `setup()` function, to absolutely avoid conflicts that would lead the MCU to crash without giving no error on the console.

Also, it's a good practice to pass the `Wire` interface from the outside, by accepting a `TwoWire`. This way you can change the physical I2C interface configuration without having to change this class in any way!

Reply:
Reading this I am not getting why would it be worse to call this function in the setup() instead of initializing the Wire directly there, other than the fact that we would need to change the class as you mentioned in the last paragraph.

Reply:
In this case, it's mainly to avoid user error. Wire functions must be called in the correct moment and in the correct order, and any mistake often leads to unexpected panics (crash with no logs). I'll list two of the possibly many scenarios that could happen.

### Multiple `begin`
Let's say that you have configured the `MotorI2C` class, and in Arduino's `setup()` you call `setupI2C()`. Then after a while you realise you also want to use a I2C sensor, so you import its library and start using. You read in it's docs that you should call `Wire.begin()` without params to initialise the com, so you execute it immediately after calling `setupI2C()`. This will lead to a panic with no logs! You can mitigate this passing the `TwoWire` interface and make sure to use two different physical interfaces, but it's much harder having to remember if the `begin()` is hidden inside another class.

### Changing pins
You need to free the pins currently used by the I2C to connect another device that could only work with those pins. So you [read the docs](https://arduino-pico.readthedocs.io/en/latest/wire.html) (let's say for the Pi Pico) and see that you should call `Wire.setSDA()` and `Wire.setSCL()` immediately before the call to `Wire.begin()` however you don't know exactly where that is called since it's hidden inside another class, and you end up putting them after `setupI2C()`. You start your program and... panic!

Even worse, if you look for the ESP32 ([docs here](https://arduino-pico.readthedocs.io/en/latest/wire.html)), you could find that to change the pins you should change the call to `Wire.begin()` and add the pins there. So you'd end up even more domain violations!